tests/test_PolarBasis2D.py

```diff
@@ -0,0 +1,127 @@
+              0.4063995 +5.21354967e-02j,  0.41872477+9.98946906e-02j,
+              0.43013599+1.38307796e-01j,  0.44144497+1.68826980e-01j,
+              0.44960099+1.94794929e-01j,  0.44872328+2.17957567e-01j,
+              0.43379428+2.36681249e-01j,  0.40485577+2.47073481e-01j,
+              0.36865853+2.45913767e-01j,  0.33597018+2.32971425e-01j,
+              0.31616666+2.10791785e-01j,  0.31313975+1.82190396e-01j],
+            [ 0.14819172-3.78171176e-03j,  0.15519289+4.75602163e-02j,
+              0.17389327+5.53498385e-02j,  0.18146449+1.37089189e-02j,
+              0.15674204-3.85815662e-02j,  0.11862477-5.15231953e-02j,
+              0.10075763-1.25689290e-02j,  0.11446016+4.53003874e-02j,
+              0.15053714+8.11915306e-02j,  0.19116871+7.99536374e-02j,
+              0.21025378+5.48466439e-02j,  0.18680217+3.31766117e-02j,
+              0.13660805+3.68947359e-02j,  0.1072859 +6.24686168e-02j,
+              0.11876919+7.93812474e-02j,  0.14075481+5.85637417e-02j,
+              0.14819172+3.78171176e-03j,  0.15519289-4.75602163e-02j,
+              0.17389327-5.53498385e-02j,  0.18146449-1.37089189e-02j,
+              0.15674204+3.85815662e-02j,  0.11862477+5.15231953e-02j,
+              0.10075763+1.25689290e-02j,  0.11446016-4.53003874e-02j,
+              0.15053714-8.11915306e-02j,  0.19116871-7.99536374e-02j,
+              0.21025378-5.48466439e-02j,  0.18680217-3.31766117e-02j,
+              0.13660805-3.68947359e-02j,  0.1072859 -6.24686168e-02j,
+              0.11876919-7.93812474e-02j,  0.14075481-5.85637417e-02j],
+            [-0.22808599-5.29338933e-02j, -0.22401193-4.33128746e-03j,
+             -0.11601473+1.35405676e-02j, -0.02110144-6.65071500e-03j,
+             -0.02886296-3.91489616e-02j, -0.05298751-1.95319479e-02j,
+             -0.04052728+5.66863498e-02j, -0.03546515+1.13544145e-01j,
+             -0.04800556+1.15828804e-01j, -0.05683092+9.72225058e-02j,
+             -0.05318826+8.54948014e-02j, -0.06674163+7.94216590e-02j,
+             -0.11467046+8.49198927e-02j, -0.12932565+1.06139634e-01j,
+             -0.1094488 +1.20159845e-01j, -0.15198775+1.02156797e-01j,
+             -0.22808599+5.29338933e-02j, -0.22401193+4.33128746e-03j,
+             -0.11601473-1.35405676e-02j, -0.02110144+6.65071500e-03j,
+             -0.02886296+3.91489616e-02j, -0.05298751+1.95319479e-02j,
+             -0.04052728-5.66863498e-02j, -0.03546515-1.13544145e-01j,
+             -0.04800556-1.15828804e-01j, -0.05683092-9.72225058e-02j,
+             -0.05318826-8.54948014e-02j, -0.06674163-7.94216590e-02j,
+             -0.11467046-8.49198927e-02j, -0.12932565-1.06139634e-01j,
+             -0.1094488 -1.20159845e-01j, -0.15198775-1.02156797e-01j]
+        ])
+
+        self.assertTrue(np.allclose(pf, result))
+
+    def testPolarBasis2DEvaluate(self):
+        v = np.zeros((4, 32))
+        v[2, 16] = 1.0
+        v[2, 0] = 1.0
+        x = self.basis.evaluate(v)
+        result = np.array([
```

Comment:
Where does this come from?

Reply:
Try to see the result in coordinate space is reasonable or not just using two symmetric nonzero points in Polar Fourier space.

Reply:
I recreated a new case for this function.